apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,669 @@
+
+### Outputs
+
+In exchange your hook should return back a JSON object with these fields set:
+
+********************************On Success********************************
+
+Return these only if your hook processed the input without errors.
+
+- `decision` A string containing the decision whether to allow authentication to move forward. Use `reject` to completely reject the verification attempt and log the user out of all active sessions. Use `continue` to use the default Supabase Auth behavior.
+- `message` The message to show the user if the decision was `reject`.
+- `should_logout_user` Whether to logout a user if a `reject` decision is issued. Has no effect when a `continue` decision is issued.
+
+Example output:
+
+```json
+{
+   "decision": "reject",
+   "message" : "You have exceeded maximum number of MFA attempts.",
+   "should_logout_user": "false"
+}
+```
+
+****************On Error****************
+
+Return this if your hook processed the input with some error.
+
+- `error` An object that contains information about the error.
+    - `http_code` A number indicating the HTTP code to be returned. If not set, the code is HTTP 500 Internal Server Error.
+    - `message` A message to be returned in the HTTP response. Must be set.
+- In general all other fields attached to the returned JSON are ignored.
+
+Example output:
+
+```jsx
+{
+  "error": {
+    "http_code": 429,
+    "message": "You can only verify a factor once very 10 seconds."
+  }
+}
+```
+
+## Custom Access Token Hook
+
+You may wish to add additional claims based on the authentication method used or add additional metadata to select users. The custom access token hook runs before a token is issued and allows you to add claims.
+
+### Inputs
+
+- `user_id`Unique identifier for the user attempting to sign in. Correlate this to the `auth.users` table.
+- `claims` Claims which are attached to the access token.
+- `authentication_method` the authentication method used to request for the access token. Possible values include:
+    - `oauth`
+    - `password`
+    - `otp`
+    - `totp`
+    - `recovery`
+    - `invite`
+    - `sso/saml`
+    - `magiclink`
+    - `email/signup`
+    - `email_change`
+    - `token_refresh`
+
+### Outputs
+
+In exchange your hook should return back a JSON object with these fields set:
+
+********************************On Success********************************
+
+Return these only if your hook processed the input without errors.
+
+- `claims` A json containing the updated claims after the hook has been run
+
+****************On Error****************
+
+Return this if your hook processed the input with some error.
+
+- `error` An object that contains information about the error.
+    - `http_code` A number indicating the HTTP code to be returned. If not set, the code is HTTP 500 Internal Server Error.
+    - `message` A message to be returned in the HTTP response. Must be set.
+- In general all other fields attached to the returned JSON are ignored.
```

Comment:
You can move this duplicate error explanation to the technical overview or the section about writing hooks.